From 84eb4381db85877a9a56b35994e6c10d43e46ebe Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 14 Jul 2014 12:04:54 +0000
Subject: seat: be a little more forgiving in the case there's no open terminal

We can end up in a situation where a seat object doesn't have a terminal
associated with it.  In that case we shouldn't crash, but continue on
with no input available for that seat.

https://bugs.freedesktop.org/show_bug.cgi?id=80553
---
diff -Naur plymouth-0.9.0/src/libply-splash-core/ply-seat.c plymouth-0.9.0.tpg/src/libply-splash-core/ply-seat.c
--- plymouth-0.9.0/src/libply-splash-core/ply-seat.c	2014-03-06 19:42:01.000000000 +0000
+++ plymouth-0.9.0.tpg/src/libply-splash-core/ply-seat.c	2014-07-15 18:22:43.201670281 +0000
@@ -153,15 +153,23 @@
       seat->keyboard = ply_keyboard_new_for_renderer (seat->renderer);
       add_pixel_displays (seat);
 
-    }
-  else
-    {
+
+  } else if (seat->terminal != NULL) {
+
       seat->keyboard = ply_keyboard_new_for_terminal (seat->terminal);
     }
-  add_text_displays (seat);
+        if (seat->terminal != NULL) {
+                add_text_displays (seat);
+        } else {
+                ply_trace ("not adding text display for seat, since seat has no associated terminal");
+        }
 
-  ply_keyboard_watch_for_input (seat->keyboard);
-  seat->keyboard_active = true;
+        if (seat->keyboard != NULL) {
+                ply_keyboard_watch_for_input (seat->keyboard);
+                seat->keyboard_active = true;
+        } else {
+                ply_trace ("not watching seat for input");
+        }
 
   return true;
 }
